src/trs/rewrite.rs

```diff
@@ -0,0 +1,128 @@
+//! Hindley-Milner Typing for First-Order Term Rewriting Systems (no abstraction)
```

Comment:
Perhaps `Polymorphically typed (Hindley-Milner) First-Order...` since we don't actually provide "typing" here.